test/PowerShellEditorServices.Test.E2E/Processes/StdioServerProcess.cs

```diff
@@ -0,0 +1,137 @@
+        /// </param>
+        /// <param name="serverStartInfo">
+        ///     A <see cref="ProcessStartInfo"/> that describes how to start the server.
+        /// </param>
+        public StdioServerProcess(ILoggerFactory loggerFactory, ProcessStartInfo serverStartInfo)
+            : base(loggerFactory)
+        {
+            if (serverStartInfo == null)
+                throw new ArgumentNullException(nameof(serverStartInfo));
+
+            _serverStartInfo = serverStartInfo;
+        }
+
+        /// <summary>
+        ///     Dispose of resources being used by the launcher.
+        /// </summary>
+        /// <param name="disposing">
+        ///     Explicit disposal?
+        /// </param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                Process serverProcess = Interlocked.Exchange(ref _serverProcess, null);
+                if (serverProcess != null)
+                {
+                    if (!serverProcess.HasExited)
+                        serverProcess.Kill();
+
+                    serverProcess.Dispose();
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Is the server running?
+        /// </summary>
+        public override bool IsRunning => !ServerExitCompletion.Task.IsCompleted;
+
+        /// <summary>
+        ///     The server's input stream.
+        /// </summary>
+        public override Stream InputStream => _serverProcess?.StandardInput?.BaseStream;
+
+        /// <summary>
+        ///     The server's output stream.
+        /// </summary>
+        public override Stream OutputStream => _serverProcess?.StandardOutput?.BaseStream;
+
+        /// <summary>
+        ///     Start or connect to the server.
+        /// </summary>
+        public override Task Start()
+        {
+            ServerExitCompletion = new TaskCompletionSource<object>();
+
+            _serverStartInfo.CreateNoWindow = true;
+            _serverStartInfo.UseShellExecute = false;
+            _serverStartInfo.RedirectStandardInput = true;
+            _serverStartInfo.RedirectStandardOutput = true;
+
+            Process serverProcess = _serverProcess = new Process
+            {
+                StartInfo = _serverStartInfo,
+                EnableRaisingEvents = true
+            };
+            serverProcess.Exited += ServerProcess_Exit;
+
+            if (!serverProcess.Start())
+                throw new InvalidOperationException("Failed to launch language server .");
+
+            ServerStartCompletion.TrySetResult(null);
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        ///     Stop or disconnect from the server.
+        /// </summary>
+        public override async Task Stop()
+        {
+            Process serverProcess = Interlocked.Exchange(ref _serverProcess, null);
+            if (serverProcess != null && !serverProcess.HasExited)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5596138)